h2o-core/src/main/java/water/util/MRUtils.java

```diff
@@ -223,6 +223,9 @@ public static Frame sampleFrameStratified(final Frame fr, Vec label, Vec weights
       numrows += sampling_ratios[i] * dist[i];
     }
     if (Float.isNaN(numrows)) {
+      Log.err("Total number of sampled rows was NaN. " +
```

Comment:
why not improve the exception message as well, this way the user has to dig through the logs to get the details which could easily be part of the exception

Reply:
The number of classes can be large and including all in the message might cause issues on the client (large JSON).

We can add information about which class was the problematic one but this most likely won't be user-actionable, this is most likely caused by a numerical instability of our math.